tests/e2e/resilience_test.go

```diff
@@ -234,6 +239,141 @@ func (tc *OperatorResilienceTestCtx) ValidateComponentsDeploymentFailure(t *test
+	crdBackup := tc.resourceBackup(t, &componentCrd)
+	t.Cleanup(func() {
+		tc.EventuallyResourceCreatedOrUpdated(
+			WithObjectToCreate(crdBackup),
+			WithCustomErrorMsg("Failed to restore CRD from backup (cleanup)"),
+		)
+	})
+
+	// Delete the CRD to simulate missing CRD
+	propagationPolicy := metav1.DeletePropagationBackground
+	tc.DeleteResource(
+		WithMinimalObject(gvk.CustomResourceDefinition, types.NamespacedName{Name: componentCrd.GetName()}),
+		WithClientDeleteOptions(
+			&client.DeleteOptions{
+				PropagationPolicy: &propagationPolicy,
+			}),
+		WithWaitForDeletion(true),
+	)
+
+	// Validate pod health and system health
+	selector := tc.getOperatorPodSelector()
+	tc.validatePodHealth(t, selector)
+
+	// Manually restore the CRD from backup
+	tc.EventuallyResourceCreatedOrUpdated(
+		WithObjectToCreate(crdBackup),
+		WithCustomErrorMsg("Failed to restore CRD from backup"),
+	)
+	tc.validateSystemHealth(t)
+}
+
+func (tc *OperatorResilienceTestCtx) ValidateRBACRestrictionHandlings(t *testing.T) {
+	t.Helper()
+
+	// Fetch the ClusterRoleBinding that exists and save a backup copy
+	crbName := tc.getControllerDeploymentName() + "-rolebinding"
+	crb := tc.FetchResource(
+		WithMinimalObject(gvk.ClusterRoleBinding, types.NamespacedName{Name: crbName}),
+	)
+
+	// Check if ClusterRoleBinding exists
+	if crb == nil {
+		t.Logf("Available clusterrolebindings: %v", tc.FetchResources(WithMinimalObject(gvk.ClusterRoleBinding, types.NamespacedName{})))
+		t.Skipf("ClusterRoleBinding %s not found, skipping RBAC restriction test", crbName)
+		return
+	}
+
+	// Save a backup copy of the CRB (deep copy)
+	crbBackup := tc.resourceBackup(t, crb)
+
+	// Add cleanup to ensure CRB is restored even if test fails
+	t.Cleanup(func() {
+		tc.EventuallyResourceCreatedOrUpdated(
+			WithObjectToCreate(crbBackup),
+			WithCustomErrorMsg("Failed to restore ClusterRoleBinding from backup (cleanup)"),
+			WithWaitForDeletion(true),
+		)
+	})
+
+	// Deleting ClusterRoleBinding to simulate RBAC restriction
+	tc.DeleteResource(
+		WithMinimalObject(gvk.ClusterRoleBinding, types.NamespacedName{Name: crbName}),
+	)
+
+	// Trigger a reconciliation that would expose RBAC issues
+	t.Log("Triggering reconciliation to expose RBAC issues...")
+	tc.EventuallyResourceCreatedOrUpdated(
+		WithMinimalObject(gvk.DataScienceCluster, tc.DataScienceClusterNamespacedName),
+		WithMutateFunc(testf.Transform(`.metadata.labels.test = "rbac-restriction"`)),
+		WithEventuallyTimeout(tc.TestTimeouts.mediumEventuallyTimeout),
+		WithEventuallyPollingInterval(tc.TestTimeouts.defaultEventuallyPollInterval),
+	)
+
+	// Check Pod health and search logs for forbidden errors
+	selector := tc.getOperatorPodSelector()
+	tc.validatePodHealth(t, selector)
+	leaderPod := tc.findLeaderPodFromLeases()
+	tc.g.Expect(leaderPod).ShouldNot(BeEmpty(), "Failed to find leader pod")
+	tc.g.Expect(tc.searchPodLogs(t, leaderPod, tc.OperatorNamespace, "forbidden")).Should(BeTrue())
```

Comment:
Same as before. I'm not sure why we need to check the logs. I'd rather check how it reacts while reconciling the DSC.